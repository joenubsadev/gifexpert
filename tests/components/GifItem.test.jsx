import { render, screen } from "@testing-library/react"
const { GifItem } = require("../../src/components/GifItem")

describe ('Pruebas en <GifItem.jsx>', () => {
    const title = "Hola titulo";
    const url = "https://holamundo.com/holatitulo"
    test('debe hacer match con snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    });

    test('imagen con url y alt indicados', () => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        // console.log ( screen.getByRole('img').src);
        // expect ( screen.getByRole('img').src).toBe( url );
        // expect ( screen.getByRole('img').alt).toBe( title );

        const { src, alt } = screen.getByRole('img')
        expect (src).toBe(url);
        expect (alt).toBe(title);
    })

    test ('debe mostrar el titulo en el componente',  () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})