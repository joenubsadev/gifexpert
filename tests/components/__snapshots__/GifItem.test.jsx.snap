// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem.jsx> debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola titulo"
      src="https://holamundo.com/holatitulo"
    />
    <p>
      Hola titulo
    </p>
  </div>
</div>
`;
